libs/ngxtension/trackby-id-prop/src/trackby-id-prop.ts

```diff
@@ -0,0 +1,42 @@
+//INSPIRED BY https://medium.com/ngconf/make-trackby-easy-to-use-a3dd5f1f733b
+import { NgForOf } from '@angular/common';
+import {
+	Directive,
+	Input,
+	Provider,
+	inject,
+	type NgIterable,
+} from '@angular/core';
+
+@Directive({
+	selector: '[ngForTrackById]',
+	standalone: true,
+})
+export class NgForTrackByIdDirective<T extends { id: string | number }> {
+	@Input() ngForOf!: NgIterable<T>;
+	private ngFor = inject(NgForOf<T>, { self: true });
+
+	constructor() {
+		this.ngFor.ngForTrackBy = (index: number, item: T) => item.id;
+	}
+}
+
+@Directive({
+	selector: '[ngForTrackByProp]',
+	standalone: true,
+})
+export class NgForTrackByPropDirective<T> {
```

Comment:
nit: what do you think of dropping Directive off of the name? Like NgFor and NgIf, they don't have the Directive appended.

Reply:
Same as above ;-)